external/storm-redis/src/main/java/org/apache/storm/redis/bolt/RedisFilterBolt.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.storm.tuple.Tuple;
+import redis.clients.jedis.GeoCoordinate;
+import redis.clients.jedis.JedisCommands;
+
+import java.util.List;
+
+/**
+ * Basic bolt for querying from Redis and filters out if key/field doesn't exist.
+ * If key/field exists on Redis, this bolt just forwards input tuple to default stream.
+ * <p/>
+ * Supported data types: STRING, HASH, SET, SORTED_SET, HYPER_LOG_LOG, GEO.
+ * <p/>
+ * Note: For STRING it checks such key exists on the key space.
+ * For HASH and SORTED_SET and GEO, it checks such field exists on that data structure.
+ * For SET and HYPER_LOG_LOG, it check such value exists on that data structure.
+ * (Note that it still refers key from tuple via RedisFilterMapper#getKeyFromTuple())
+ * In order to apply checking this to SET, you need to input additional key this case.
+ * <p/>
+ * Note2: If you want to just query about existence of key regardless of actual data type,
+ * specify STRING to data type of RedisFilterMapper.
+ */
+public class RedisFilterBolt extends AbstractRedisBolt {
+    private final RedisFilterMapper filterMapper;
+    private final RedisDataTypeDescription.RedisDataType dataType;
+    private final String additionalKey;
+
+    /**
+     * Constructor for single Redis environment (JedisPool)
+     * @param config configuration for initializing JedisPool
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisPoolConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * Constructor for Redis Cluster environment (JedisCluster)
+     * @param config configuration for initializing JedisCluster
+     * @param filterMapper mapper containing which datatype, query key that Bolt uses
+     */
+    public RedisFilterBolt(JedisClusterConfig config, RedisFilterMapper filterMapper) {
+        super(config);
+
+        this.filterMapper = filterMapper;
+
+        RedisDataTypeDescription dataTypeDescription = filterMapper.getDataTypeDescription();
+        this.dataType = dataTypeDescription.getDataType();
+        this.additionalKey = dataTypeDescription.getAdditionalKey();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void execute(Tuple input) {
+        String key = filterMapper.getKeyFromTuple(input);
+
+        boolean found;
+        JedisCommands jedisCommand = null;
+        try {
+            jedisCommand = getInstance();
+
+            switch (dataType) {
+                case STRING:
+                    found = jedisCommand.exists(key);
+                    break;
+
+                case SET:
+                    if (additionalKey == null) {
```

Comment:
since data type is known at the time of construction, can this check be moved to constructor itself? Failing fast is preferred.